test/main.lua

```diff
@@ -303,8 +312,18 @@ local function save_test_status(status)
     json.encode_file(status, STATUS_FILE)
 end
 
+-- causes scripts to be reloaded the next time they are reqscript()'d. this
+-- allows scripts that change their behavior based on the value of
+-- dfhack.internal.IN_TEST to return to non-test behavior after tests are run.
+local function invalidate_scripts()
+    for k,_ in pairs(dfhack.internal.scripts) do
+        dfhack.internal.scripts[k] = nil
```

Comment:
This would affect scripts that rely on global variables persisting across runs - values of those would be cleared permanently. Not a likely issue, but see `devel/lua-example` for a simple example of a script that would be reset by this logic.

Reply:
that is true. is that desired behavior or undesired, though?  If you're running tests, is it more useful to keep or reset global state?